src/Microsoft.OData.Edm/Csdl/Parsing/CsdlJsonParser.cs

```diff
@@ -0,0 +1,335 @@
+                        includeAlias = propertyValue.ParseAsString(context);
+                        break;
+
+                    case "$Namespace":
+                        // The value of $Namespace is a string containing the namespace of the included schema
+                        includeNamespace = propertyValue.ParseAsString(context);
+                        break;
+
+                    default:
+                        // The item objects MAY contain annotations.
+                        // However, EdmInclude does not supported yet. So skip
+                        context.ReportError(EdmErrorCode.JsonUnexpectedElement, Strings.CsdlJsonParser_UnexpectedJsonMember(context.Path, propertyValue.ValueKind));
+                        break;
+                }
+            });
+
+            return new EdmInclude(includeAlias, includeNamespace);
+        }
+
+        /// <summary>
+        /// Parse the <see cref="JsonElement"/> to a <see cref="IEdmIncludeAnnotations"/>.
+        /// </summary>
+        /// <param name="element">The input JSON element.</param>
+        /// <param name="context">The parser context.</param>
+        /// <returns>null or parsed <see cref="IEdmIncludeAnnotations"/>.</returns>
+        internal static IEdmIncludeAnnotations ParseIncludeAnnotations(JsonElement element, JsonParserContext context)
+        {
+            // Each item in $IncludeAnnotations is an object.
+            if (!element.ValidateValueKind(JsonValueKind.Object, context))
+            {
+                return null;
+            }
+
+            string termNamespace = null;
+            string qualifier = null;
+            string targetNamespace = null;
+            element.ParseAsObject(context, (propertyName, propertyValue) =>
+            {
+                // Array items are objects that MUST contain the member $Namespace and MAY contain the member $Alias.
+                switch (propertyName)
+                {
+                    case "$TermNamespace":
+                        // The value of $TermNamespace is a namespace.
+                        termNamespace = propertyValue.ParseAsString(context);
+                        break;
+
+                    case "$Qualifier":
+                        // The value of $Qualifier is a simple identifier.
+                        qualifier = propertyValue.ParseAsString(context);
+                        break;
+
+                    case "$TargetNamespace":
+                        // The value of $TargetNamespace is a namespace.
+                        targetNamespace = propertyValue.ParseAsString(context);
+                        break;
+
+                    default:
+                        // The item objects MAY contain annotations. However, IEdmIncludeAnnotations doesn't support to have annotations.
+                        context.ReportError(EdmErrorCode.JsonUnexpectedElement,

```

Comment:
>ReportError [](start = 32, length = 11)

These are non-fatal errors, right? We report that we didn't parse it, but we don't fail the parse?  We should never fail the parse due to not being able to understand an annotation.

Reply:
Yes. 
JsonUnsupportedElement & JsonUnexpectedElement are both for non-fatal errors.

---
In reply to: [443024023](https://github.com/OData/odata.net/pull/1775#discussion_r443024023) [](ancestors = 443024023)

Reply:
Great; thanks.

---
In reply to: [444556340](https://github.com/OData/odata.net/pull/1775#discussion_r444556340) [](ancestors = 444556340,443024023)